keel-sql/src/main/kotlin/com/netflix/spinnaker/keel/sql/SqlResourceRepository.kt

```diff
@@ -121,35 +125,36 @@ open class SqlResourceRepository(
   }
 
   // todo: this is not retryable due to overall repository structure: https://github.com/spinnaker/keel/issues/740
-  override fun store(resource: Resource<*>) {
-    val uid = jooq.select(RESOURCE.UID)
+  override fun store(resource: Resource<*>): Resource<*> {
+    val uid = randomUID().toString()
+    val version = jooq
+      .select(
+        coalesce(
+          max(RESOURCE.VERSION).plus(1),
+          value(1)
+        )
+      )
       .from(RESOURCE)
       .where(RESOURCE.ID.eq(resource.id))
-      .fetchOne(RESOURCE.UID)
-      ?: randomUID().toString()
+      .fetchOneInto(Int::class.java)
 
-    val updatePairs = mapOf(
-      RESOURCE.KIND to resource.kind,
-      RESOURCE.ID to resource.id,
-      RESOURCE.SPEC to objectMapper.writeValueAsString(resource.spec),
-      RESOURCE.APPLICATION to resource.application
-    )
-    val insertPairs = updatePairs + (RESOURCE.UID to uid)
-    jooq.insertInto(
-      RESOURCE,
-      *insertPairs.keys.toTypedArray()
-    )
-      .values(*insertPairs.values.toTypedArray())
-      .onDuplicateKeyUpdate()
-      .set(updatePairs)
+    jooq.insertInto(RESOURCE)
+      .set(RESOURCE.UID, uid)
+      .set(RESOURCE.KIND, resource.kind.toString())
+      .set(RESOURCE.ID, resource.id)
+      .set(RESOURCE.VERSION, version)
+      .set(RESOURCE.SPEC, objectMapper.writeValueAsString(resource.spec))
+      .set(RESOURCE.APPLICATION, resource.application)
       .execute()
```

Comment:
We no longer update resources, just insert a new version

Reply:
Should we be worried about the size of the table after this change, and whether to cap the max number of versions for a resource (in a future PR, if at all)?

Reply:
Still gonna be a lot smaller than the `event` table. I'm not worried just yet. But, yes I did suggest in the proposal doc that we could prune older versions after a while.